XamlAnimatedGif/Animator.cs

```diff
@@ -503,24 +503,38 @@ private void ClearArea(Int32Rect rect)
 
         private async Task<Stream> GetIndexStreamAsync(GifFrame frame, CancellationToken cancellationToken)
         {
-            var data = frame.ImageData;
-            cancellationToken.ThrowIfCancellationRequested();
-            _sourceStream.Seek(data.CompressedDataStartOffset, SeekOrigin.Begin);
-            using (var ms = new MemoryStream(_indexStreamBuffer))
+            try
+            {
+                var data = frame.ImageData;
+                cancellationToken.ThrowIfCancellationRequested();
+                _sourceStream.Seek(data.CompressedDataStartOffset, SeekOrigin.Begin);
+                using (var ms = new MemoryStream(_indexStreamBuffer))
+                {
+                    await GifHelpers.CopyDataBlocksToStreamAsync(_sourceStream, ms, cancellationToken).ConfigureAwait(false);
+                }
+                var lzwStream = new LzwDecompressStream(_indexStreamBuffer, data.LzwMinimumCodeSize);
+                return lzwStream;
+            }
+            catch (ObjectDisposedException)
             {
-                await GifHelpers.CopyDataBlocksToStreamAsync(_sourceStream, ms, cancellationToken).ConfigureAwait(false);
+                return null;
```

Comment:
I don't think it's that simple. The code that calls `GetIndexStreamAsync` doesn't expect it to return null. In `RenderFrameAsync`, it's called when `_cacheFrameDataInMemory` is false. And in the code that follows, if `_cacheFrameDataInMemory` is false, `indexStream` is assumed to not be null. If it is, `RenderFrameAsync` will throw.